d2/src/main/java/com/linkedin/d2/jmx/D2ClientJmxManager.java

```diff
@@ -129,17 +129,17 @@ public void onServicePropertiesRemoval(LoadBalancerStateItem<ServiceProperties>
 
       private String getClusterInfoJmxName(String clusterName)
       {
-        return String.format("%s-ClusterInfo", clusterName);
+        return String.format("%s-%s-ClusterInfo", _prefix, clusterName);
       }
 
       private String getServicePropertiesJmxName(String serviceName)
       {
-        return String.format("%s-ServiceProperties", serviceName);
+        return String.format("%s-%s-ServiceProperties", _prefix, serviceName);
       }
 
       private String getLoadBalancerStrategyJmxName(String serviceName, String scheme)
       {
-        return serviceName + "-" + scheme + "-LoadBalancerStrategy";
+        return String.format("%s-%s-%s-LoadBalancerStrategy", _prefix, serviceName, scheme);
       }
```

Comment:
!!!!!
This will break existing alerts! You need to preserve the old metric, otherwise it could break the monitoring for certain apps.

Reply:
as we discussed, used the prefix only in xDS LB metric names to keep the existing ZK LB metric names intact for backward compatibility (even though that was a bug before which should've used the prefix).

Reply:
updated the names based on primary and secondary source according to the dual read mode.